test/e2e/cypress/e2e/hosts_overview.cy.js

```diff
@@ -170,4 +170,41 @@ context('Hosts Overview', () => {
       });
     });
   });
+
+  describe('Deregistration', () => {
+    describe('Clean-up buttons should be visible only when needed', () => {
+      const heartbeatTargets = agents();
+      const failingHost = heartbeatTargets.shift();
+      before(() => {
+        cy.task('startAgentHeartbeat', heartbeatTargets);
+      });
+
+      it('should hide all Clean-up buttons when heartbeat starts except the first one', () => {
+        cy.get('tr:nth-child(1) > td:nth-child(9)').should(
+          'contain.text',
+          'Clean up'
+        );
+
+        for (let i = 2; i < 11; i++) {
+          cy.get(
+            `:nth-child(${i}) > .w-48 > .bg-white > .text-jungle-green-500`
+          ).should('not.exist');
+        }
+      });
+
+      it('should allow to deregister a host after clean up confirmation', () => {
```

Comment:
Should this test belong to the current `describe`. It is not about the button visualization. This already runs a deregistration, and most probably we want to have more things in this, like restoration and other things